packages/lodestar-beacon-state-transition/src/fast/block/processVoluntaryExit.ts

```diff
@@ -43,10 +43,10 @@ export function processVoluntaryExit(
   if (verifySignature) {
     const domain = getDomain(config, state, DomainType.VOLUNTARY_EXIT, voluntaryExit.epoch);
     const signingRoot = computeSigningRoot(config, config.types.VoluntaryExit, voluntaryExit, domain);
-    if (!verify(
-      validator.pubkey.valueOf() as Uint8Array,
+    const pubkey = epochCtx.index2pubkey[voluntaryExit.validatorIndex];
```

Comment:
Is it possible for `voluntaryExit.validatorIndex` to not exist in `epochCtx.index2pubkey`? If so, should we guard against it or not necessary?

Reply:
Nope validator must exists for it to exit, state transition checks this before we check signature here